test/functional/users_controller_test.rb

```diff
@@ -201,6 +201,22 @@ def setup
     assert_equal User.find(user.id).bio, 'Hello, there!'
   end
 
+  test 'username should not be updated1' do
+    user = users(:bob)
+    UserSession.create(user)
+    user.username = 'newval'
+    user.save
+    assert_equal user.username, 'newval'
```

Comment:
You can check the behaviour in rails console OR try
`assert_equal user.username, 'n' ` it will show in the terminal what is the expected vs actual output.
That can help you.